drivers/clock_control/clock_control_nrf.c

```diff
@@ -463,16 +512,19 @@ void z_nrf_clock_control_lf_on(enum nrf_lfclk_start_mode start_mode)
 		__ASSERT_NO_MSG(err >= 0);
 	}
 
-	switch (start_mode) {
-	case NRF_LFCLK_START_MODE_SPINWAIT_STABLE:
-		lfclk_spinwait(CLOCK_CONTROL_NRF_K32SRC);
-		break;
+	/* In case of simulated board leave immediately. */
+	if (IS_ENABLED(CONFIG_SOC_SERIES_BSIM_NRFXX)) {
+		return;
+	}
 
-	case NRF_LFCLK_START_MODE_SPINWAIT_RUNNING:
-		lfclk_spinwait(NRF_CLOCK_LFCLK_RC);
+	switch (start_mode) {
+	case CLOCK_CONTROL_NRF_LF_START_AVAILABLE:
+		/* fall through. */
+	case CLOCK_CONTROL_NRF_LF_START_STABLE:
```

Comment:
The comment here is not needed. There is no code between those two cases, so there will be no warning here. It's not a not a "true fall through". ;-)